### DISCUSSION TITLE
Помогите пожалуйста. Второй день мучаюсь

### DISCUSSION THREAD

**Discussion by qINiKIp** [vvip-68/GyverPanelWiFi#317] [Задавайте вопросы]
In file included from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/third_party/arduinojson/json.h:55,
                 from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/json.h:3,
                 from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/screenmap.h:11,
                 from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/engine_events.h:6,
                 from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/cled_controller.h:17,
                 from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/cpixel_ledcontroller.h:16,
                 from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/controller.h:7,
                 from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/FastLED.h:65,
                 from C:\Users\red_a\OneDrive\Рабочий стол\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\a_def_hard.h:431,
                 from C:\Users\red_a\OneDrive\Рабочий стол\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\GyverPanelWiFi_v1.13.ino:253:
c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/third_party/arduinojson/json.hpp:8127:2: error: #error ARDUINOJSON_SLOT_OFFSET_SIZE has been removed, use ARDUINOJSON_SLOT_ID_SIZE instead
 8127 | #error ARDUINOJSON_SLOT_OFFSET_SIZE has been removed, use ARDUINOJSON_SLOT_ID_SIZE instead
      |  ^~~~~
exit status 1

Compilation error: exit status 1

**Comment by qINiKIp** [comment]
Сори если что-то не так отправил (с гит хабом не знаком)

**Reply by vvip-68** [reply]
А если просто инструкцию хотя бы в начале основного файла прочитать? Какие версии нужно ставить, какие не нужно... С какими будет работать, а с какими вот так - даже не соберется...

**Reply by qINiKIp** [reply]
Установил нужную библиотеку теперь стало вот так:

In file included from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/FastLED.h:79,
                 from D:\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\a_def_hard.h:431,
                 from D:\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\GyverPanelWiFi_v1.13.ino:253:
c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src/fastspi.h:157:23: note: '#pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output'
  157 | #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src\FastLED.h:79,
                 from c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src\five_bit_hd_gamma.cpp:1:
c:\Users\red_a\OneDrive\Documents\Arduino\libraries\FastLED\src\fastspi.h:157:23: note: '#pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output'
  157 | #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
. Variables and constants in RAM (global, static), used 42124 / 80192 bytes (52%)
║   SEGMENT  BYTES    DESCRIPTION
╠══ DATA     1760     initialized variables
╠══ RODATA   6508     constants       
╚══ BSS      33856    zeroed variables
. Instruction RAM (IRAM_ATTR, ICACHE_RAM_ATTR), used 61099 / 65536 bytes (93%)
║   SEGMENT  BYTES    DESCRIPTION
╠══ ICACHE   32768    reserved space for flash instruction cache
╚══ IRAM     28331    code in IRAM    
. Code in flash (default, ICACHE_FLASH_ATTR), used 552900 / 1048576 bytes (52%)
║   SEGMENT  BYTES    DESCRIPTION
╚══ IROM     552900   code in flash

Вроде все загружается но если подключаю пин D4 к ленте не соединяя минусы ленты и платы то начинают моргать рандомные диоды. Если же соединить минусы то ничего не горит совсем.
В монитор порта при перезагрузке плата отправляет вот это:
�EtI�<Eh��4h`��h�hbr��tB@��J�E�5T�	(��	�կ+,��׍-Q!�u�a���\�%��ņ+5�	����R#�)�D9
��!�8�)�5h��
���+);�+p[�9�9�)1����1k�1#d����
)������?E�����:m��i�������ǵG@�.

**Reply by vvip-68** [reply]
1. Без соединения минуса ничего работать не будет - сигнал идет по двум проводам.
2. Если ничего по подключениями не меняли -на D4 обычно вешается кнопка, вывод на ленту идет с пина D2/ Если меняли - разбирайтесь что вы там насоединяли.
3. Чтобы в мониторе порта было что-то осознанное, его скорость должна быть синхронизировано со скоростью, указанной в скетче. В скетче это 115200 бод.

**Reply by qINiKIp** [reply]
> 1. Без соединения минуса ничего работать не будет - сигнал идет по двум проводам.
> 2. Если ничего по подключениями не меняли -на D4 обычно вешается кнопка, вывод на ленту идет с пина D2/ Если меняли - разбирайтесь что вы там насоединяли.
> 3. Чтобы в мониторе порта было что-то осознанное, его скорость должна быть синхронизировано со скоростью, указанной в скетче. В скетче это 115200 бод.

Извиняюсь. Не D4, а D2. Но ситуацию с минусами осталась. Подключаю DIN ленты к D2 моргают случайные диоды. Как только соединяю минусы все тухнет. 

Загружал скетч для проверки ленты:

#define NUM_LEDS 264 // кол-во всех светодиодов в испытуемой ленте
#include "FastLED.h"
#define PIN 6
CRGB leds[NUM_LEDS];
byte counter;
void setup() {
FastLED.addLeds<WS2811, PIN, GRB>(leds, NUM_LEDS).setCorrection( TypicalLEDStrip );
FastLED.setBrightness(50);
pinMode(13, OUTPUT);
}
void loop() {
for (int i = 0; i < NUM_LEDS; i++ ) { // от 0 до первой трети
leds[i] = CHSV(counter + i * 2, 255, 255); // HSV. Увеличивать HUE (цвет)
// умножение i уменьшает шаг радуги
}
counter++; // counter меняется от 0 до 255 (тип данных byte)
FastLED.show();
delay(5); // скорость движения радуги
}

С ним лента работает.

**Reply by shkolin1973** [reply]
У вас точно стоит правильная версия Arduino IDE, ядро, и библиотеки (из проекта и правильная FastLed ?)

**Reply by shkolin1973** [reply]
Выложите куда ни будь вашу версию проекта я попробую её скачать и зашить.

**Reply by shkolin1973** [reply]
У вас папка с проектом случайно не на "Рабочем столе" лежит? Папку проекта лучше в корень диска C: класть. Arduino IDE не любит длинные пути к файлам.

**Reply by vvip-68** [reply]
@shkolin1973 - у него проект то собрался и загрузился. Значит не в Arduino IDE дело. У него вывода на матрицу нет. Значит насоединял что-то не так.

**Reply by shkolin1973** [reply]
А что там не так можно соединить :-)))))   +/- 5В на ленту и контроллер и D2- на Din ленты через резистор. Он же пишет, что тестовый скетч работает правильно.

**Reply by vvip-68** [reply]
@shkolin1973 , о, вы не представляете, насколько изобретательны пользователи. Даже читая инструкцию и "я все сделал как написано", все равно что-то делают не так. Встречался мне один уникум, который на NodeMCU землю припаял не к GND, а к выводу CMD и долго не могли понять почему ничего не заводится. Но - "сделал всё по инструкции". И примеров масса.

**Reply by shkolin1973** [reply]
Такое бывает. Может у человека зрение слабое. А на ESP32 вообще контакт рядом с +5В подписан как GND, а это вовсе не GND, а как раз CMD. :-))))  Может на NodeMCU так же

**Reply by vvip-68** [reply]
@shkolin1973, ну, набросайте тогда идей, что еще могло пойти не так? Скетч, как я понял скомпилился и загрузился в плату. Всё сделано по инструкции. Настроек никаких не менял (хотя путается в показаниях D2/D4, и библиотеки использовал сначала вовсе не те версии, про которые написано в инструкции). Ну и в чем тогда причина? У меня идеи кончились. Хрустальный шар затуманился...

**Reply by shkolin1973** [reply]
Уже всё написали, пусть проверяет. 
[qINiKIp] сбросьте сюда скрины где видно версию установленного ядра микроконтроллера и FastLed.
Настройки микроконтроллера перед записью.
Аааааа.... При записи выбирете вариант "Всё стереть перед записью" 
![Скриншот 11-12-2024 150628](https://github.com/user-attachments/assets/19be8b28-aa8c-4871-9971-7694d0ccc639)

И выложите отредактированный под себя (то, что вы заливаете) проект. Я попробую его залить и проверить.
Я так понял вы на ESP8266 делаете

**Reply by vvip-68** [reply]
Если скетч запускается, то в мониторе порта при старте микроконтроллера  пишется и тип МК, и версии ядра / FastLED и размеры матриц. V 1.14 еще пишется к каким пинам что подключено, в 1.13 - не помню пишется или нет.

Ну и в любом случае потыкать проводочком входа на ленту поочередно в D2 и в D4 - не сложно же? Вряд ли какие-то другие варианты.

**Reply by shkolin1973** [reply]
В 1,13 пишется версия ядра и FastLED, пины нет. Я управляющий пин я проверял светодиодом, на нем он мерцает.

**Reply by qINiKIp** [reply]
Всем спасибо за ответы. До этого грузил на плату Wemos D1 mini. Нашел старую NodeMCU (думал она не рабочая) Загрузил тот же скетч и все заработало.
